### DISCUSSION TITLE
Access metastring in Fence component to highlight line numbers

### DISCUSSION THREAD

**Discussion by ian-whitestone** [markdoc/markdoc#241] [Q&A] [answered]
Greetings!

I am trying to add the ability to highlight specific lines of code in my `Fence` component, following [this guide](https://prince.dev/highlight-with-react). Given code block, like:

<img width="399" alt="CleanShot 2022-10-07 at 11 56 20@2x" src="https://user-images.githubusercontent.com/11096727/194620219-cbe81223-738f-4be4-9ba8-d4b2b57247ea.png">

I need the `Fence` component to get access to the metastring `'{1,3-4}'` so I can parse out the line numbers and style them accordingly. I tried adding it as a new attribute in my `nodes.js`:

```javascript
// nodes.js
const nodes = {
    ...
  fence: {
    render: Fence,
    attributes: {
      language: {
        type: String,
      },
      metastring: {
        type: String,
      },
    },
  },
  ...
}
```

And then passing that to the `Fence` component:

```javascript
// Fence.jsx


export function Fence({ children, language, metastring }) {
  console.log(language)
  console.log(metastring)
  const shouldHighlightLine = calculateLinesToHighlight('{1,2,3-4}');

...
```

But `metastring` is always `undefined`. Any ideas how I can pass through an additional attribute? Thanks!

**Comment by mfix-stripe** [comment]
@ian-whitestone two things: for one, there is a condition in our parser where the `language` attribute is only captured if there is no space before the content.

If you remove the space in your example, then those values will show up in the `language` attribute of the `fence` node. 

Here is a playground that demonstrates how this works: https://markdoc.dev/sandbox?mode=ast&c=YGBgezEsMy00fQpzZWxlY3QgLi4uCmBgYA%253D%253D

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/62121649/194625898-bf52f799-4719-4977-9267-d55d962aae78.png">

**Comment by rpaul-stripe** [comment, marked as answer]
To pass a custom attribute to a fenced code block, use the following syntax:

~~~
```sql {% attributename="1,3-4" %}
code goes here
```
~~~

**Comment by ian-whitestone** [comment]
Thanks @mfix-stripe and @rpaul-stripe for the quick responses! Using @rpaul-stripe 's solution worked.
